enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/utils/default-embed-setting.ts

```diff
@@ -1,26 +1,45 @@
 import { match } from "ts-pattern";
 
+import type {
+  DashboardEmbedOptions,
+  ExplorationEmbedOptions,
+  QuestionEmbedOptions,
+} from "metabase-enterprise/embedding_iframe_sdk/types/embed";
+
 import type { SdkIframeEmbedSetupExperience } from "../types";
 
 export const getDefaultSdkIframeEmbedSettings = (
   type: SdkIframeEmbedSetupExperience,
   defaultEntityId: string | number,
 ) =>
   match(type)
-    .with("dashboard", () => ({
-      dashboardId: defaultEntityId,
-      isDrillThroughEnabled: true,
-      withDownloads: false,
-      withTitle: true,
-    }))
-    .with("chart", () => ({
-      questionId: defaultEntityId,
-      isDrillThroughEnabled: true,
-      withDownloads: false,
-      withTitle: true,
-    }))
-    .with("exploration", () => ({
-      template: "exploration",
-      isSaveEnabled: true,
-    }))
+    .with(
+      "dashboard",
+      (): DashboardEmbedOptions => ({
```

Comment:
Adds the embed option types so we can use `DEFAULT_SDK_IFRAME_EMBED_SETTINGS` below in preview and initial state.